### PR TITLE
Wi 60 b/e edit user profile adam

### PR DESCRIPTION
**Pull request by Adamyyyy**

### CODE REVIEW

File: src/admin/schema/user.schema.ts [hunk 1]
```diff
@@ -28,6 +24,9 @@ export class User {
 
   @Prop({ default: "user", type: String, enum: ["user", "admin"] })
   role: string;
+
+  @Prop({ default: "default_avatar.png", type: String })
+  avatar: string;
```
Comment (Chanran) [inline src/admin/schema/user.schema.ts:29]: avatarPath or avatarName？

File: src/editUser/editUser.controller.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  UseGuards,
+  UseInterceptors,
+} from "@nestjs/common";
+import { S3Client, PutObjectCommand } from "@aws-sdk/client-s3";
+
+import { EditUserService } from "./editUser.service";
+import { JwtAuthGuard } from "src/auth/guards/jwt-auth.guards";
+import { EditUserDto } from "./dto/edit-user.dto";
+
+const bucketName = process.env.BUCKET_NAME;
```
Comment (Chanran) [inline src/editUser/editUser.controller.ts:24]: Put all the process.env.xxx in a file named "config.ts".  And read the env variable in config.ts.

File: src/editUser/editUser.controller.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+  @Get("me")
+  async me(@Req() req: Request, @Res() res: Response) {
+    const userEmail = req.user["email"];
+    const user = await this.editUserService.findByEmail(userEmail);
+    return res.status(HttpStatus.OK).json({
+      message: "success",
+      data: {
+        firstName: user.firstName,
+        lastName: user.lastName,
+        avatar: `https://${bucketName}.s3.${bucketRegion}.amazonaws.com/${user.avatar}`,
```
Comment (Chanran) [inline src/editUser/editUser.controller.ts:64]: Define a variable to express avatar link.

File: src/editUser/editUser.controller.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+      credentials: {
+        accessKeyId: accessKey,
+        secretAccessKey: secretAccessKey,
+      },
+    });
+
+    await s3Clinet.send(
+      new PutObjectCommand({
+        Bucket: bucketName,
+        Body: fileBuffer,
```
Comment (Chanran) [inline src/editUser/editUser.controller.ts:106]: Will the original file upload to S3 too?

File: src/editUser/editUser.controller.ts [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        Key: filename,
+        ContentType: file.mimetype,
+      })
+    );
+
+    // get user email from jwt token
+    const userEmail = req.user["email"];
+
+    // update avatar in db
+    const newAvatarFileName = await this.editUserService.updateAvatarByEmail(
```
Comment (Chanran) [inline src/editUser/editUser.controller.ts:116]: I think you should store the avatar link to db as well

File: src/editUser/editUser.service.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    userEmail: string,
+    dto: EditUserDto
+  ): Promise<{ firstName: string; lastName: string }> {
+    const updateUser = await this.userModel.findOneAndUpdate(
+      { email: userEmail },
+      { ...dto },
+      { new: true }
+    );
+
+    if (!updateUser) throw new NotFoundException();
```
Comment (Chanran) [inline src/editUser/editUser.service.ts:26]: use {} to wrap content of if

File: src/editUser/editUser.controller.ts (hunk 4 above)
Comment (Adamyyyy) [reply src/editUser/editUser.controller.ts:112]: Only the resized avatar file will be uploaded to S3.

**Review by JeremyyyLiu — APPROVED** [review]
Go Merge

**Pull request merged**